api/src/main/java/com/github/retrooper/packetevents/protocol/packettype/PacketType.java

```diff
@@ -431,22 +484,22 @@ public static class Configuration {
 
         public enum Client implements PacketTypeCommon, ServerBoundPacket {
 
-            CLIENT_SETTINGS,
-            PLUGIN_MESSAGE,
-            CONFIGURATION_END_ACK,
-            KEEP_ALIVE,
-            PONG,
-            RESOURCE_PACK_STATUS,
+            CLIENT_SETTINGS(WrapperConfigClientSettings.class),
+            PLUGIN_MESSAGE(WrapperConfigClientPluginMessage.class),
+            CONFIGURATION_END_ACK(WrapperConfigClientConfigurationEndAck.class),
+            KEEP_ALIVE(WrapperConfigClientKeepAlive.class),
+            PONG(WrapperConfigClientPong.class),
+            RESOURCE_PACK_STATUS(WrapperConfigClientResourcePackStatus.class),
 
             // Added in 1.20.5
-            COOKIE_RESPONSE,
-            SELECT_KNOWN_PACKS;
+            COOKIE_RESPONSE(WrapperConfigClientCookieResponse.class),
+            SELECT_KNOWN_PACKS(WrapperConfigClientSelectKnownPacks.class);
 
             private static int INDEX = 0;
             private static final Map<Byte, Map<Integer, PacketTypeCommon>> PACKET_TYPE_ID_MAP = new HashMap<>();
             private final int[] ids;
 
-            Client() {
+            Client(@Nullable Class<?> clazz) {
```

Comment:
```suggestion
            Client(@Nullable Class<? extends PacketWrapper<?>> clazz) {
```

Also, this doesn't seem to be saved + exposed as a field